lib/rubygems/webauthn_listener/response.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+##
+# The WebauthnListener Response class is used by the WebauthnListener to print
+# the specified response to the Gem host using the provided socket. It also closes
+# the socket after printing the response.
+#
+# Types of response classes:
+#   - ResponseOk
+#   - ResponseNoContent
+#   - ResponseBadRequest
+#   - ResponseNotFound
+#   - ResponseMethodNotAllowed
+#
+# Example:
+#   socket = TCPSocket.new(host, port)
+#   Gem::WebauthnListener::ResponseOk.send(socket, host)
+#
+
+class Gem::WebauthnListener
+  class Response
```

Comment:
Should we use the [`HTTPResponseFactory`](https://github.com/rubygems/rubygems/blob/1424c26861c30719afd7519db65cda9f75d4d7fa/test/rubygems/utilities.rb#L166-L187) somehow?

Reply:
Oh I think we can... instead of building a raw string response, it might be better to create a HTTPResponse object and use that to send to the Gem host 👍  I'll try it out